samcli/lib/observability/cw_logs/cw_log_group_provider.py

```diff
@@ -144,9 +144,17 @@ def for_step_functions(
             log_group_arn = logging_destination.get("cloudWatchLogsLogGroup", {}).get("logGroupArn")
             LOG.debug("Log group ARN: %s", log_group_arn)
             if log_group_arn:
-                log_group_arn_parts = log_group_arn.split(":")
-                log_group_name = log_group_arn_parts[6]
-                return str(log_group_name)
+                if ":" in log_group_arn and len(log_group_arn.split(":")) > 6:
+                    log_group_arn_parts = log_group_arn.split(":")
+                    log_group_name = log_group_arn_parts[6]
+                    return str(log_group_name)
+
+                LOG.debug(
```

Comment:
Seems like a LOG.warning to me.

Reply:
Yeah I am in between `debug` vs `warning` here. I don't want this to be annoying. But since this will be printed once we can change this to `warning` I believe 🤔